pySDC/projects/PinTSimE/estimation_check.py

```diff
@@ -202,84 +221,309 @@ def accuracy_check(dt_list, problem, sweeper, V_ref, cwd='./'):
+            e_plt1 = e_ax.plot(times_SE_adapt, e_emb_SE_adapt, 'k--', label=r'SE+A - $\epsilon_{emb}$')
+            e_plt2 = e_ax.plot(times_adapt, e_emb_adapt, 'g--', label=r'A - $\epsilon_{emb}$')
             e_ax.set_yscale('log', base=10)
             e_ax.set_ylim(1e-16, 1e-7)
             e_ax.tick_params(labelsize=6)
 
             lines = dt1 + e_plt1 + dt2 + e_plt2
             labels = [l.get_label() for l in lines]
 
-            ax_acc[count_ax].legend(lines, labels, frameon=False, fontsize=6, loc='upper left')
+            ax_acc[count_ax].legend(lines, labels, frameon=False, fontsize=6, loc='upper right')
 
         else:
-            ax_acc.set_title(r'$\Delta t$={}'.format(dt_item))
-            dt1 = ax_acc.plot([v[0] for v in dt_TT_val], [v[1] for v in dt_TT_val], 'ko-', label=r'SE+A - $\Delta t$')
-            dt2 = ax_acc.plot([v[0] for v in dt_FT_val], [v[1] for v in dt_FT_val], 'go-', label=r'A - $\Delta t$')
-            ax_acc.axvline(x=t_switch_adapt, linestyle='--', linewidth=0.5, color='r', label='Switch')
+            ax_acc.set_title(r'$\Delta t_\mathrm{initial}$=%s' % dt_item)
+            dt1 = ax_acc.plot(
+                [v[0] for v in dt_SE_adapt_val],
+                [v[1] for v in dt_SE_adapt_val],
+                'ko-',
+                label=r'SE+A - $\Delta t_\mathrm{adapt}$',
+            )
+            dt2 = ax_acc.plot(
+                [v[0] for v in dt_adapt_val],
+                [v[1] for v in dt_adapt_val],
+                'go-',
+                label=r'A - $\Delta t_\mathrm{adapt}$',
+            )
+            ax_acc.axvline(x=t_switch_SE_adapt, linestyle='--', linewidth=0.5, color='r', label='Switch')
+            ax_acc.tick_params(axis='both', which='major', labelsize=6)
             ax_acc.set_xlabel('Time', fontsize=6)
-            ax_acc.set_ylabel(r'$Delta t_{adapted}$', fontsize=6)
+            ax_acc.set_ylabel(r'$Delta t_\mathrm{adapt}$', fontsize=6)
 
             e_ax = ax_acc.twinx()
-            e_plt1 = e_ax.plot(times_TT, e_emb_TT, 'k--', label=r'SE+A - $\epsilon_{emb}$')
-            e_plt2 = e_ax.plot(times_FT, e_emb_FT, 'g--', label=r'A - $\epsilon_{emb}$')
+            e_plt1 = e_ax.plot(times_SE_adapt, e_emb_SE_adapt, 'k--', label=r'SE+A - $\epsilon_{emb}$')
+            e_plt2 = e_ax.plot(times_adapt, e_emb_adapt, 'g--', label=r'A - $\epsilon_{emb}$')
             e_ax.set_yscale('log', base=10)
             e_ax.tick_params(labelsize=6)
 
             lines = dt1 + e_plt1 + dt2 + e_plt2
             labels = [l.get_label() for l in lines]
 
-            ax_acc.legend(lines, labels, frameon=False, fontsize=6, loc='upper left')
+            ax_acc.legend(lines, labels, frameon=False, fontsize=6, loc='upper right')
 
         count_ax += 1
 
     fig_acc.savefig('data/embedded_error_adaptivity_{}.png'.format(sweeper), dpi=300, bbox_inches='tight')
     plt_helper.plt.close(fig_acc)
 
 
+def check_solution(stats, dt, problem, use_adaptivity, use_switch_estimator):
```

Comment:
Maybe this function can be unified with the one from the model files? I am guessing one step size is the same? Then you could import this function from there and use it for testing.